src/pass/onnx_rewrite.cpp

```diff
@@ -17,6 +17,46 @@
 using namespace mlir;
 
 namespace {
+
+// Check whether an ArrayAttr contains non-zero values or not.
+bool hasNonZeroInArrayAttr(ArrayAttr attrs) {
+  bool allZeros = true;
+  if (attrs) {
+    for (auto attr: attrs.getValue()) {
+      if (attr.cast<IntegerAttr>().getInt() > 0) {
+        allZeros = false;
+        break;
+      }
+    }
+  }
+  return !allZeros;
+}
+
+// Create an ArrayAttr of IntergerAttr(s) of zero values.
+// This function is used for padding attribute in MaxPoolSingleOut.
+ArrayAttr createArrayAttrOfZeroWithTrail(
```

Comment:
"Trail" suggests to me that the zeros are added at the end. I would just call this method createExtendedArrayAttrOfZeros